dao/application_dao.go

```diff
@@ -79,3 +80,57 @@ func (a *ApplicationDaoImpl) Delete(id *int64) error {
 func (a *ApplicationDaoImpl) Tenant() *int64 {
 	return a.TenantID
 }
+
+func (a *ApplicationDaoImpl) BulkMessage(id *int64) (map[string]interface{}, error) {
+	application := &m.Application{ID: *id}
+	resource := DB.Preload("Source.Tenant").Preload("Source.Applications.Tenant").Preload("Source.Endpoints.Tenant").Find(&application)
+
+	if resource.Error != nil {
+		return nil, resource.Error
+	}
+
+	bulkMessage := map[string]interface{}{}
+	bulkMessage["source"] = application.Source.ToEvent()
+
+	endpoints := make([]m.EndpointEvent, len(application.Source.Endpoints))
+	for i, endpoint := range application.Source.Endpoints {
+		endpoints[i] = *endpoint.ToEvent()
+	}
+
+	bulkMessage["endpoints"] = endpoints
+
+	applications := make([]m.ApplicationEvent, len(application.Source.Applications))
+	for i, application := range application.Source.Applications {
+		applications[i] = *application.ToEvent()
+	}
+
+	bulkMessage["applications"] = applications
+
+	bulkMessage["authentications"] = []m.Authentication{}
+	bulkMessage["application_authentications"] = []m.ApplicationAuthenticationEvent{}
+
+	return bulkMessage, nil
+}
+
+func (a *ApplicationDaoImpl) FetchAndUpdateBy(id *int64, updateAttributes map[string]interface{}) error {
+	src, err := a.GetById(id)
+	if err != nil {
+		err = DB.Model(src).Updates(updateAttributes).Error
+	}
+
+	return err
```

Comment:
This can be slightly refactored here:

```suggestion
	result := DB.Model(&m.Application{ID: *id}).Updates(updateAttributes)
	if result.RowsEffected == 0 {
		return fmt.Errorf("application not found %v", id)
	}
	
	return nil
```

We do not need to fetch the app from the db before the update, this should generate the same sql - we can key off of whether any rows were touched in the update - it returns 1 even if no fields get updated.

Reply:
Done in https://github.com/RedHatInsights/sources-api-go/pull/49/commits/6a5d9ab142e8cabf74a243d747d19f5701580088